### PR TITLE
CEDS-2171 Integrate Customs Declaration Information service

### PR DESCRIPTION
**Pull request by PauloPortugal**
with @apodi & @PauloPortugal

### CODE REVIEW

File: app/config/AppConfig.scala [hunk 1]
```diff
@@ -62,6 +62,27 @@ class AppConfig @Inject()(
   lazy val classificationHelpUrl = loadConfig("urls.classificationHelp")
   lazy val ecicsToolUrl = loadConfig("urls.ecicsTool")
 
+  lazy val customsDeclarationsInformation = servicesConfig.baseUrl("customs-declarations-information")
+  lazy val fetchMrnStatus = servicesConfig.getConfString(
```
Comment (SteveSugden) [inline app/config/AppConfig.scala]: Could maybe have a helper method to get services config or throw exception - like `loadConfig`?

File: app/models/dis/parsers/MrnStatusParser.scala [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    documents.foreach { doc =>
+      val id = (doc \ XmlTags.id).text
+      val typeCode = (doc \ XmlTags.typeCode).text
+      previousDocuments :+= PreviousDocument(id, typeCode)
+    }
+    previousDocuments
+  }
+
+  private def timeFormatter(zonedDateTime: String): String = timeFormatter(
+    ZonedDateTime.parse(zonedDateTime, DateTimeFormatter.ofPattern("yyyyMMddHHmmssz"))
```
Comment (SteveSugden) [inline app/models/dis/parsers/MrnStatusParser.scala]: I'm fairly sure it's OK to create local constants for these DateTimeFormatters - they are thread-safe

File: app/models/dis/parsers/MrnStatusParser.scala [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+  }
+
+  private def previousDocuments(documents: NodeSeq): Seq[PreviousDocument] = {
+    var previousDocuments: Seq[PreviousDocument] = Seq.empty
+    documents.foreach { doc =>
+      val id = (doc \ XmlTags.id).text
+      val typeCode = (doc \ XmlTags.typeCode).text
+      previousDocuments :+= PreviousDocument(id, typeCode)
+    }
+    previousDocuments
```
Comment (PatrykRudnicki) [inline app/models/dis/parsers/MrnStatusParser.scala]: Please not use `var`

```
documents.map { doc =>
      val id = (doc \ XmlTags.id).text
      val typeCode = (doc \ XmlTags.typeCode).text
      PreviousDocument(id, typeCode)
    }
```

Comment (alsahid1993) [reply app/models/dis/parsers/MrnStatusParser.scala]: this has been addressed

File: app/models/dis/parsers/MrnStatusParser.scala (hunk 2 above)
Comment (alsahid1993) [reply app/models/dis/parsers/MrnStatusParser.scala]: this has been addressed

File: app/config/AppConfig.scala (hunk 1 above)
Comment (alsahid1993) [reply app/config/AppConfig.scala]: as discussed with Steve, this may not be necessary

Comment (SteveSugden) [reply app/config/AppConfig.scala]: True - it was not a blocker

Comment (bambuchaAdm) [reply app/config/AppConfig.scala]: I would check here is address is correct - ends with slash

**Pull request merged**